src/testing/creation.rs

```diff
@@ -0,0 +1,311 @@
+        .units_per_creator
+        .iter()
+        .all(|r| *r >= (max_round - 1) as usize));
+    finish(killers, handles).await;
+}
+
+// Disconnect test
+// This test starts with 7 creators. After 25 rounds 2 of them are disconnected and reconnected
+// again after the rest gets to round 50.
+// Then it is checked if 5 creators achieve round 75 and rest at least round 73.
+#[tokio::test(flavor = "multi_thread", worker_threads = 3)]
+async fn disconnected_creators_should_create_dag() {
+    let n_members = NodeCount(7);
+    let max_round: Round = 25;
+
+    let (mut test_controller, killers, handles, mut units_from_controller, units_for_creators) =
+        setup_test(n_members);
+    loop {
+        futures::select! {
+            _ = test_controller.control_until(max_round).fuse() => break,
+            unit = units_from_controller.next() => match unit {
+                Some(unit) => for units_for_creator in &units_for_creators {
+                    units_for_creator.unbounded_send(unit.clone()).expect("Channel to creator should be open");
+                },
+                None => panic!("Channel from controller should be open."),
+            }
+        }
+    }
+    let max_round: Round = 50;
+    let mut dropped_units = Vec::new();
+    loop {
+        futures::select! {
+            _ = test_controller.control_until(max_round).fuse() => break,
+            unit = units_from_controller.next() => match unit {
+                Some(unit) => {
+                    for units_for_creator in units_for_creators.iter().skip(2) {
+                        units_for_creator.unbounded_send(unit.clone()).expect("Channel to creator should be open");
+                    }
+                    dropped_units.push(unit);
+                },
+                None => panic!("Channel from controller should be open."),
+            }
+        }
+    }
+    let max_round: Round = 75;
+    for unit in dropped_units {
+        for units_for_creator in units_for_creators.iter().take(2) {
+            units_for_creator
+                .unbounded_send(unit.clone())
+                .expect("Channel to creator should be open");
+        }
+    }
+    loop {
+        futures::select! {
+            _ = test_controller.control_until(max_round).fuse() => break,
+            unit = units_from_controller.next() => match unit {
+                Some(unit) => for units_for_creator in &units_for_creators {
+                    units_for_creator.unbounded_send(unit.clone()).expect("Channel to creator should be open");
+                },
+                None => panic!("Channel from controller should be open."),
+            }
+        }
+    }
+    assert!(test_controller
+        .units_per_creator
+        .iter()
+        .all(|r| *r >= (max_round - 3) as usize));
+    assert!(
+        test_controller
+            .units_per_creator
+            .iter()
+            .map(
+                // There should be an easier way to do this...
```

Comment:
I don't have a better idea, maybe `.fold()`? Still marking this line, coz this comment should be deleted before merge. Hope other review will help with this

Reply:
Yeah, I was hoping a reviewer would figure something better out and then I could delete this. I think `fold` is marginally better, but I just had an idea to use `filter`+`count` which looks acceptable to me.